libs/sql-parser/src/main/java/io/crate/sql/SqlFormatter.java

```diff
@@ -1219,6 +1223,64 @@ public Void visitSetSessionAuthorizationStatement(SetSessionAuthorizationStateme
             return null;
         }
 
+        @Override
+        public Void visitDeclare(Declare declare, Integer indent) {
+            builder
+                .append("DECLARE ")
+                .append(declare.cursorName())
+                .append(" ");
+            if (declare.binary()) {
+                builder.append("BINARY ");
+            }
+            builder.append(declare.scroll() ? "SCROLL " : "NO SCROLL ");
+            builder.append("CURSOR ");
+            builder.append(declare.hold() == Declare.Hold.WITH ? "WITH HOLD " : "WITHOUT HOLD ");
+            builder.append("FOR ");
+            declare.query().accept(this, indent);
+            return null;
+        }
+
+        @Override
+        public Void visitFetch(Fetch fetch, Integer context) {
+            builder.append("FETCH ");
+            ScrollMode scrollMode = fetch.scrollMode();
+            long count = fetch.count();
+            if (scrollMode == ScrollMode.ABSOLUTE) {
+                if (count == 1) {
+                    builder.append("FIRST ");
+                } else if (fetch.count() == -1) {
+                    builder.append("LAST ");
+                } else {
+                    builder.append("ABSOLUTE ");
+                    builder.append(fetch.count());
+                    builder.append(" ");
+                }
+            } else {
+                if (count >= 0) {
+                    builder.append("FORWARD ");
+                } else {
+                    builder.append("BACKWARD ");
+                }
+                if (count == Long.MAX_VALUE || count == Long.MAX_VALUE * -1) {
```

Comment:
why not use `Long.MIN_VALUE` here?
```
System.out.println(Long.MIN_VALUE);
System.out.println(Long.MAX_VALUE * -1);
=>
-9223372036854775808
-9223372036854775807
```
Is it specified by the CURSOR requirements?

Reply:
Makes it easier to use the same value as sentinel for ALL